File: 73-set-matrix-zeroes/73-set-matrix-zeroes.java
class Solution {
    class Pair{
        int r,c;
        Pair(int row,int col){
            r=row;
            c=col;
        }
    }
    public void setZeroes(int[][] matrix) {
        Queue<Pair> q=new ArrayDeque<>();
        for(int i=0;i<matrix.length;i++){
            for(int j=0;j<matrix[0].length;j++){
                if(matrix[i][j]==0){
                    q.add(new Pair(i,j));
                }
            }
        }
        
        while(!q.isEmpty()){
            Pair index=q.poll();
            for(int i=0;i<matrix.length;i++){
                matrix[i][index.c]=0;
            }
            for(int i=0;i<matrix[0].length;i++){
                matrix[index.r][i]=0;
            }
        }
    }
}

File: 73-set-matrix-zeroes/73-set-matrix-zeroes.py
class Solution(object):
    def setZeroes(self, matrix):
        R=len(matrix)
        C=len(matrix[0])
        isCol = False
        for i in range(R):
            if matrix[i][0]==0 :
                isCol=True
            for j in range(1,C):
                if matrix[i][j]==0:
                    matrix[i][0]=0
                    matrix[0][j]=0
            
           
            
        for i in range(1,R):
            for j in range(1,C):
                if(matrix[i][0]==0 or matrix[0][j]==0):
                    matrix[i][j]=0;
        
        if matrix[0][0]==0:
            for j in range(C):
                    matrix[0][j]=0
            
        if isCol:
            for i in range(R):
                matrix[i][0]=0